i18n fails missing ./langs folder

### Version
nuxt-i18n: 6.13.3
nuxt: 2.14.0

### Nuxt configuration
#### [mode](https://nuxtjs.org/api/configuration-mode): <!--universal is the default, please insert X where appropriate -->
 - [x] universal

### Nuxt-i18n configuration
```
export default {
  modules: ['nuxt-i18n'],
  i18n: {
    locales: [
      {
        name: 'German',
        code: 'de',
        iso: 'de-DE',
        file: 'de-DE.json',
      },
    ],
    defaultLocale: 'de',
    lazy: true,
    langDir: 'locales/',
  },
}

```

### Steps to reproduce

Use above config with only one language.

### What is Expected?

It just works™ as it worked until yesterday. :)

### What is actually happening?
With the above config, i18n expects a `langs` folder. However, since only one language is present no `langs` folder is created, thus the build fails.

```
ERROR in ./dist/nuxt-i18n/utils.js
Module not found: Error: Can't resolve './langs' in '/my/path/to/project/dist/nuxt-i18n'
 @ ./dist/nuxt-i18n/utils.js 87:19-89:36
 @ ./dist/nuxt-i18n/plugin.main.js
 @ ./dist/index.js
 @ ./dist/client.js
 @ multi ./dist/client.js
```

Before one might ask the question why we only use one language: More languages are a requirement for the future, so we're already preparing the project from the get-go. Also i18n is a great way to work with language files rather than hardcoding stuff. ;)

--- a/src/index.js
+++ b/src/index.js
@@ -31,22 +31,32 @@ module.exports = function (userOptions) {
 
   if (!Object.values(STRATEGIES).includes(options.strategy)) {
     // eslint-disable-next-line no-console
-    console.error('[' + options.MODULE_NAME + '] Invalid "strategy" option "' + options.strategy + '" (must be one of: ' + Object.values(STRATEGIES).join(', ') + ').')
+    console.error('[' + MODULE_NAME + '] Invalid "strategy" option "' + options.strategy + '" (must be one of: ' + Object.values(STRATEGIES).join(', ') + ').')
     return
   }
 
   let defaultLangFile
+  let hasNonDefaultLangFiles = false
 
   // Copy lang files to the build directory.
   if (options.langDir) {
     if (!options.locales.length || typeof options.locales[0] === 'string') {
-      console.warn('[' + options.MODULE_NAME + '] When using "langDir" option, the "locales" option must be a list of objects')
+      console.error('[' + MODULE_NAME + '] When using "langDir" option, the "locales" option must be a list of objects')
     }
 
     const uniqueFiles = new Set(options.locales.map(locale => locale.file))
 
     if (options.defaultLocale) {
-      defaultLangFile = options.locales.find(locale => locale.code === options.defaultLocale).file
+      const defaultLocaleObject = options.locales.find(locale => locale.code === options.defaultLocale)
+      if (defaultLocaleObject) {
+        if (defaultLocaleObject.file) {
+          defaultLangFile = defaultLocaleObject.file
+        } else {
+          console.error(`[${MODULE_NAME}] Default locale is missing the "file" property (required when using "langDir" option)`)
+        }
+      } else {
+        console.error(`[${MODULE_NAME}] Default locale is set to "${options.defaultLocale}" but no locale with that "code" was found`)
+      }
     }
 
     for (const file of uniqueFiles) {
@@ -55,6 +65,9 @@ module.exports = function (userOptions) {
         src: resolve(this.options.srcDir, options.langDir, file),
         fileName: join(ROOT_DIR, (isUsingDefaultLangFile ? 'default-lang' : 'langs'), file)
       })
+      if (!isUsingDefaultLangFile) {
+        hasNonDefaultLangFiles = true
+      }
     }
   }
 
@@ -72,6 +85,7 @@ module.exports = function (userOptions) {
     STRATEGIES,
     COMPONENT_OPTIONS_KEY,
     defaultLangFile,
+    hasNonDefaultLangFiles,
     localeCodes,
     trailingSlash
   }
@@ -104,12 +118,12 @@ module.exports = function (userOptions) {
     }
   } else if (options.differentDomains) {
     // eslint-disable-next-line no-console
-    console.warn('[' + options.MODULE_NAME + '] The `differentDomains` option and `no_prefix` strategy are not compatible. Change strategy or disable `differentDomains` option.')
+    console.warn('[' + MODULE_NAME + '] The `differentDomains` option and `no_prefix` strategy are not compatible. Change strategy or disable `differentDomains` option.')
   }
 
   if ('forwardedHost' in options) {
     // eslint-disable-next-line no-console
-    console.warn('[' + options.MODULE_NAME + '] The `forwardedHost` option is deprecated. You can safely remove it. See: https://github.com/nuxt-community/i18n-module/pull/630.')
+    console.warn('[' + MODULE_NAME + '] The `forwardedHost` option is deprecated. You can safely remove it. See: https://github.com/nuxt-community/i18n-module/pull/630.')
   }
 
   const templatesPath = join(__dirname, '/templates')

--- a/src/templates/utils.js
+++ b/src/templates/utils.js
@@ -40,7 +40,9 @@ export async function loadLanguageAsync (context, locale) {
             if (file === defaultLangFile) {
               langFileModule = defaultLangModule
             } else {
+              /* <% if (options.hasNonDefaultLangFiles) { %> */
               langFileModule = await import(/* webpackChunkName: "lang-[request]" */ `./langs/${file}`)
+              /* <% } %> */
             }
             const getter = langFileModule.default || langFileModule
             messages = typeof getter === 'function' ? await Promise.resolve(getter(context, locale)) : getter

--- a/test/module.test.js
+++ b/test/module.test.js
@@ -1262,6 +1262,49 @@ describe('en-US locale with no explicit default locale (issue #628)', () => {
   })
 })
 
+describe('lazy with single (default) locale', () => {
+  /** @type {Nuxt} */
+  let nuxt
+
+  beforeAll(async () => {
+    const override = {
+      i18n: {
+        lazy: true,
+        langDir: 'lang/',
+        defaultLocale: 'en-US',
+        vueI18n: {
+          messages: null
+        }
+      }
+    }
+
+    const localConfig = loadConfig(__dirname, 'basic', override, { merge: true })
+
+    // Override after merging options to avoid arrays being merged.
+    localConfig.i18n.locales = [
+      {
+        code: 'en-US',
+        iso: 'en-US',
+        name: 'English',
+        file: 'en-US.js'
+      }
+    ]
+
+    nuxt = (await setup(localConfig)).nuxt
+  })
+
+  afterAll(async () => {
+    await nuxt.close()
+  })
+
+  // Issue https://github.com/nuxt-community/i18n-module/issues/824
+  test('page loads without errors (issue #824)', async () => {
+    const html = await get('/')
+    const dom = getDom(html)
+    expect(dom.querySelector('body')?.textContent).toContain('page: Homepage')
+  })
+})
+
 describe('external vue-i18n configuration', () => {
   /** @type {Nuxt} */
   let nuxt

